matching-core/src/main/java/com/graphhopper/matching/ViterbiMatchEntry.java

```diff
@@ -0,0 +1,316 @@
+/**
+ * Copyright (C) 2015-2016, BMW Car IT GmbH and BMW AG
+ * Author: Stefan Holder (stefan.holder@bmw.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.graphhopper.matching;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.bmw.hmm.Transition;
+import com.carrotsearch.hppc.procedures.IntProcedure;
+import com.graphhopper.coll.GHBitSet;
+import com.graphhopper.coll.GHIntHashSet;
+import com.graphhopper.coll.GHTBitSet;
+import com.graphhopper.routing.Path;
+import com.graphhopper.routing.QueryGraph;
+import com.graphhopper.routing.VirtualEdgeIteratorState;
+import com.graphhopper.routing.util.EdgeFilter;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.index.LocationIndexTree;
+import com.graphhopper.storage.index.QueryResult;
+import com.graphhopper.util.DistanceCalc;
+import com.graphhopper.util.DistancePlaneProjection;
+import com.graphhopper.util.EdgeExplorer;
+import com.graphhopper.util.EdgeIterator;
+import com.graphhopper.util.EdgeIteratorState;
+
+/**
+ * This is a wrapper around an input MatchEntry (which is itself a wrapped around an input
+ * GPXEntry), which contains additional information to work with the Viterbi algorithm (via
+ * hmm-lib). For example, it stores the candidates, emission/transition probabilities, etc.
+ * 
+ * @author Stefan Holder
+ * @author kodonnell
+ */
+public class ViterbiMatchEntry {
+
+    /**
+     * DistanceCalc used for e.g. detecting how far candidates are from the original point.
+     * TODO: needs to be DistancePlanProject to be consistent with prior behavior - DistanceCalcEarth fails.
```

Comment:
This is not a big problem here IMO. As the tests says: return at least 4-5, 4-7. So more precise means just less results here we should probably make the tests like set.contains() instead of equals to fix this .